ITISFoundation/osparc-simcore
# ✨♻️ (⚠️ devops) Remove 5gb limit when uploading data via nodeports

## ⚠️ devops

@mrnicegyu11 @Surfict 
Environment variables changes:
- renamed `R_CLONE_S3_PROVIDER` to `R_CLONE_PROVIDER`
- Below env vars became mandatory for `director-v2`(**make sure they exist on the deployment**):
    - S3_ENDPOINT
    - S3_ACCESS_KEY
    - S3_SECRET_KEY
    - S3_BUCKET_NAME
    - S3_SECURE

## What do these changes do?



To avoid hitting a wall when uploading more than 5GB via AWS S3 `rclone` is used to upload data to ports.

- if `nodeports_v2` is provided with an `r_clone_settings` instance it will use `rclone` to upload data to s3
- the `s3_link` used by r_clone is generated by `storage`
- when the upload is complete a handler to update the metadata in `storage` is called (using the same code as the previous method)
- in case the `rclone` upload fails, metadata entries will be removed (trying to reduce trash)
- `dynamic-sidecar` now has a mandatory `r_clone_settings` fields.

BONUS:
- when dynamic-sidecar start outputs directories no longer get uploaded (disabled file system events while running  `docker-compose up`)

## Related issue/s



- requires https://github.com/ITISFoundation/osparc-simcore/pull/3004
- https://github.com/ITISFoundation/osparc-issues/issues/617
## How to test



- start locally and run a `jupyter-math:2.0.5` service. 
- add files to one of the outputs
- looking at the logs of the associated `dynamic-sidecar` you will see `r_clone` being used
- also in the logs of the `storage` service calls to the metadata endpoint will be issued to: `GET v0/locations/0/files/{s3_object}/s3/link` and `PATCH v0/locations/0/files/_S3_OBJECT_/metadata` 

## Checklist



- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
